exonum/src/node/state.rs

```diff
@@ -865,6 +925,43 @@ impl State {
         }
     }
 
+    /// Finds unknown transactions in the block.
+    pub fn create_incomplete_block(
+        &mut self,
+        msg: &BlockResponse,
+        txs: &MapIndex<&&Snapshot, Hash, RawMessage>,
+        txs_pool: &KeySetIndex<&&Snapshot, Hash>,
+    ) -> Result<bool, failure::Error> {
+        if let Some(ref incomplete_block) = self.incomplete_block {
```

Comment:
Could it make sense to move this check to `handle_block` before heavyweight operations (e.g., signature verification), and ensure that `self.incomplete_block` is `None` here?